### PR TITLE
AuthTokens have a minimum name of 1 character

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Testing insert of null on all dbs

## Related Issue
https://github.com/owncloud/enterprise/issues/2269

## How Has This Been Tested?
- [x] unit tests
- [x] manual testing

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by PVince81** [comment]
moving to triage, slightly old

**Comment by tomneedham** [comment]
@PVince81 How should we resolve this issue on Oracle? Prefix all the token names? Don't allow (new) empty named tokens (needs UI and backend work)?

**Comment by DeepDiver1975** [comment]
fixes https://github.com/owncloud/core/issues/29709

**Comment by DeepDiver1975** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! This pull request is being rebased automatically. Please DO NOT push while rebase is in progress or your changes would be lost!

**Comment by ownclouders** [comment]
Automated rebase was successful!

File: lib/private/Authentication/Token/DefaultToken.php [hunk 1]
```diff
@@ -133,4 +131,24 @@ public function setLastCheck($time) {
...
+		if (strlen($name) < 1) {
+			throw new \InvalidArgumentException();
+		}
+		return parent::setName($name);
+	}
+
+	/**
+	 * @param string $loginName
+	 */
+	public function  setLoginName($loginName) {
```
Comment (PVince81) [inline lib/private/Authentication/Token/DefaultToken.php]: double space

File: lib/private/Authentication/Token/DefaultToken.php [hunk 2]
```diff
@@ -133,4 +131,24 @@ public function setLastCheck($time) {
 		return parent::setLastCheck($time);
 	}
 
+	/**
+	 * @param string $name
+	 */
```
Comment (PVince81) [inline lib/private/Authentication/Token/DefaultToken.php]: missing doc for return value

is there even any ?

File: lib/private/Authentication/Token/DefaultToken.php [hunk 3]
```diff
@@ -133,4 +131,24 @@ public function setLastCheck($time) {
...
+	public function setName($name) {
+		if (strlen($name) < 1) {
+			throw new \InvalidArgumentException();
+		}
+		return parent::setName($name);
+	}
+
+	/**
+	 * @param string $loginName
+	 */
```
Comment (PVince81) [inline lib/private/Authentication/Token/DefaultToken.php]: PHPDoc for return value

**Review by PVince81 — APPROVED** [review]
:+1: code looks good

File: lib/private/Authentication/Token/DefaultToken.php (hunk 2 above)
Comment (DeepDiver1975) [reply lib/private/Authentication/Token/DefaultToken.php]: no return

**Pull request merged**